bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/resource/TeaserResourceWrapper.java

```diff
@@ -0,0 +1,80 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.resource;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceWrapper;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.wrappers.ValueMapDecorator;
+import org.apache.sling.models.annotations.Exporter;
+import org.jetbrains.annotations.NotNull;
+
+import com.adobe.cq.export.json.ExporterConstants;
+
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME , extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class TeaserResourceWrapper extends ResourceWrapper {
+
+    private ValueMap valueMap;
+    private String resourceType;
+
+    public TeaserResourceWrapper(@NotNull Resource resource, @NotNull String resourceType, @NotNull Map<String, String> inheritedProperties) {
```

Comment:
I'd rename `inheritedProperties` to `overriddenProperties` and `resourceType` to `overriddenResourceType`

Reply:
Addressed in https://github.com/adobe/aem-core-wcm-components/pull/1574/commits/6fa9ea079386bd95a8bad46ca676b62ee4d5bec2